### PR TITLE
cmakelists fixes and clarity improvements

### PR DESCRIPTION
**Pull request by James-Mart**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -210,39 +210,36 @@ set(ADMIN_DIR services/user/AdminSys/ui:AdminSys_js)
 set(COMMON_LIB_DIR services/user/CommonSys/common/packages/common-lib/:CommonSysCommonLib_js)
 set(COMMON_LOADER_DIR services/user/CommonSys/common/packages/wasm-loader/:CommonSysLoaderLib_js)
 
-function(add_js_target ENTRY OUTPUT_FILE)
-    string(REGEX REPLACE "^([^:]+):([^:]+)$" \\1 PATH ${ENTRY})
-    string(REGEX REPLACE "^([^:]+):([^:]+)$" \\2 DEP ${ENTRY})
+function(add_js_target ROW OUTPUT_FILE)
+    string(REGEX REPLACE "^([^:]+):([^:]+)$" \\1 PATH ${ROW})
+    string(REGEX REPLACE "^([^:]+):([^:]+)$" \\2 TargetName ${ROW})
```
Comment (brandonfancher) [inline CMakeLists.txt:215]: Why `TargetName` and not `TARGET_NAME` like other variable names? Is it just a hint that the values are in PascalCase? (`CommonSys_js`, `TokenSys_js`, etc.)?

File: CMakeLists.txt [hunk 2]
```diff
@@ -210,39 +210,36 @@ set(ADMIN_DIR services/user/AdminSys/ui:AdminSys_js)
...
+    set(OUTPUT_FILEPATH ${CMAKE_CURRENT_SOURCE_DIR}/${PATH}/dist/${OUTPUT_FILE})
```
Comment (brandonfancher) [inline CMakeLists.txt:216]: What did we decide on the merits of hard-coding `/dist` in this?

File: CMakeLists.txt [hunk 3]
```diff
@@ -210,39 +210,36 @@ set(ADMIN_DIR services/user/AdminSys/ui:AdminSys_js)
...
 endfunction()
 
-list(TRANSFORM JS_DIRS REPLACE "^[^:]+:([^:]+)$" \\1 OUTPUT_VARIABLE JS_DEPS)
-file(GLOB common LIST_DIRECTORIES false ${CMAKE_CURRENT_SOURCE_DIR}/services/user/CommonSys/common/resources/*)
+# Static (not built) resource dependencies
+file(GLOB common-misc-resources LIST_DIRECTORIES false ${CMAKE_CURRENT_SOURCE_DIR}/services/user/CommonSys/common/resources/*)
 file(GLOB common-fonts LIST_DIRECTORIES false ${CMAKE_CURRENT_SOURCE_DIR}/services/user/CommonSys/common/resources/fonts/*)
 
-# TODO: Can this be a dist folder, i.e., a build product?
-file(GLOB common-loader LIST_DIRECTORIES false ${CMAKE_CURRENT_SOURCE_DIR}/services/user/CommonSys/common/packages/wasm-loader/dist/*)
```
Comment (brandonfancher) [inline CMakeLists.txt:236]: I take it we have confirmation that loader builds without this superfluous line, eh?

**Review by brandonfancher — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (James-Mart) [reply CMakeLists.txt:215]: Yeah it's sort of a convention that target names are PascalCase. My thinking was that, even though it's a variable, it contains the name of a target and so I wanted to indicate that by giving the variable PascalCase.

But... upon reflection, I don't actually like this. It's a variable, should be caps.

File: CMakeLists.txt (hunk 3 above)
Comment (James-Mart) [reply CMakeLists.txt:236]: If anything the intention here was that other things would correctly be dependent on the loader, but it's being done incorrectly. I used the _DEP variable to ensure things properly depend on the loader.

File: CMakeLists.txt (hunk 2 above)
Comment (James-Mart) [reply CMakeLists.txt:216]: Works for now until we have output files that don't live in a `dist` directory. At which point we can change the parameters to OUTPUT_FILE_SUBPATH and pass in `dist/index.html` etc.

File: CMakeLists.txt (hunk 3 above)
Comment (brandonfancher) [reply CMakeLists.txt:236]: I see. And it's a few lines later where we build the loader (240/237). Cool.

File: CMakeLists.txt (hunk 1 above)
Comment (James-Mart) [reply CMakeLists.txt]: fixed

**Comment by James-Mart** [comment]
Closes https://github.com/gofractally/psibase/issues/627

**Pull request merged**